pennylane/kernels/cost_functions.py

```diff
@@ -175,3 +176,59 @@ def circuit(x1, x2):
         rescale_class_labels=rescale_class_labels,
         normalize=True,
     )
+
+
+def task_model_alignment(
+    N,
+    task_weights,
+    kernel_evals,
+):
+    r"""Task-model alignment for a given kernel function as proposed in [].
+
+    This function measures how much of the target function producing a supervised dataset is
+    captured in the kernel's first :math:`N` eigenvalues. More colloquially, it measures how much the basis functions
+    available to the kernel are used by the target function that we seek to learn with the kernel.
+
+    Let :math:`w_k` be the task weights for the target function that produced a supervised dataset as specified
+    by the :doc:`pennylane.kernels.utils.task_weights` function, and :math:`\lambda_k` the eigenvalues of the kernel
+    used to compute these weights. The task-model alignment is defined as:
+
+    .. math::
+
+        C(l) = \frac{\sum_{k \leq N} w_k^2 \lambda_k}{\sum_{k} w_k^2 \lambda_k}.
+
+    Evidence put forward in Ref [] and other papers shows that this alignment is closely related to the generalization
+    error of a kernel method.
+
+    Args:
+        N (int): compute the task-model alignment for components up this index; has to be smaller or equal to the
```

Comment:
this needs to be updated with `kernel_evals` and `task_weights`

Reply:
Thanks :) The docstrings are a mess